### PR TITLE
Add plantlike_rooted coral

### LINKED ISSUE #2040: Improve corals
This is something that needs doing but i'm too busy to.

We now have the 'plantlike rooted' drawtype so plantlike corals can now be used if rooted on a cubic coral reef node.
~~More colours of the cubic 'brain corals' could be added.~~ EDIT i think we have enough of those now, best add plantlike rooted corals.
If schematics are used, to create the schematic file edit https://github.com/minetest-mods/saveschems/blob/master/init.lua#L967 and run that mod in any world, the .mts file will appear in the folder.
For textures i prefer those from the 'seacoral' mod in the 'sea' modpack by Topywo, i'm not suggesting all 6 have to be used, less is probably better.
https://forum.minetest.net/viewtopic.php?f=11&t=4627&hilit=sea+modpack

screenshot_20171106_173019

seacoral_coralaqua

seacoral_coralcyan

seacoral_corallime

seacoral_coralmagenta

seacoral_coralredviolet

seacoral_coralskyblue

### PR DESCRIPTION
**Pull request by Ezhh**
This:
- Adds three new plantlike rooted corals (green, pink and cyan) using textures from the sea modpack.
- Removes the existing coral schematic.
- Switches coral placement to simple single node placements.

This is very much WIP.

In its current form placement looks like this:
coral
Please note that noise needs to be adjusted and in this example I've set it to place huge amounts of coral just so we can get an idea of what it might look like.

To do:
- Better noise values.
- Fix any other problems.

Closes #2040

### CODE REVIEW

**Comment by SmallJoker** [comment]
Now that's a very lovely decoration for the sea grounds. I'm looking forward to have this feature in MTG.

File: mods/default/nodes.lua [hunk 1]
```diff
@@ -1910,6 +1910,162 @@ minetest.register_node("default:sand_with_kelp", {
...
+	selection_box = {
+		type = "fixed",
+		fixed = {
+				{-0.5, -0.5, -0.5, 0.5, 0.5, 0.5},
+				{-4/16, 0.5, -4/16, 4/16, 1.5, 4/16},
+		},
+	},
+	drop = "",
+	node_dig_prediction = "default:coral_skeleton",
+	node_placement_prediction = "air",
```
[the remark this answers is not in the document]
Comment (paramat) [reply mods/default/nodes.lua]: In testing i get a momentary bubble of air when a placement fails.

**Comment by paramat** [comment]
Coral textures need to be indexed alpha.
In testing digging and placement are fine, not sure what's best for placement prediction.
Will work on noise params next.

**Comment by paramat** [comment]
```
		sidelen = 4,
		noise_params = {
			offset = -6,
			scale = 4,
			spread = {x = 100, y = 100, z = 100},
			seed = 7013,
			octaves = 3,
			persist = 1,
		},
```
Note sidelen 4.
This is fairly similar to previous distribution and rarity.

**Comment by jastevenson303** [comment]
I'm neutral.

**Comment by Ezhh** [comment]
Using the noise values suggested by @paramat, there are some very nice patches such as:
c1

But also found a weird square patch I am not so keen on. Unsure how common this might be.
c2

**Comment by Ezhh** [comment]
Not really WIP anymore, but still open to suggestions for changing drops and groups for the coral.

**Comment by paramat** [comment]
Yes the noise is tricky, the square area is due to the 3rd octave of noise spiking up. I need to tune more, but i'm thinking merge anyway, tune later?

**Comment by paramat** [comment]
Well, i'm neutral, you can merge if you want, or wait, i'll get working on tuning noise.

**Comment by Ezhh** [comment]
I'm fine with merging and then you can look at adjusting more later. I only found that one area in the screenshot so far that didn't look good to me, so it's possible that it's rare.

**Comment by paramat** [comment]
'cracky = 3' seems maybe a little too much effort for thin fragile coral structures, being equal to a full stone node.
Looking at papyrus and other plants, how about 'snappy = 3'?

**Comment by Ezhh** [comment]
That's what I am leaning more toward, it's just a bit odd when you have the combined structures due to the nature of the drawtype.

**Comment by paramat** [comment]
Ok. but when dug only the top extension is dug.

**Comment by Ezhh** [comment]
Yes, why I'm leaning toward it. Will change.

**Comment by paramat** [comment]
coralnoise2

Finer detail noise. Persistence 0.7 instead of 1 so shapes are a combination of a larger variety of scales. The shapes seem more interesting instead of squarish blobs.
Note the map is all hot biomes so this is the distribution within hot biomes.
```
		sidelen = 4,
		noise_params = {
			offset = -4,
			scale = 4,
			spread = {x = 50, y = 50, z = 50},
			seed = 7013,
			octaves = 3,
			persist = 0.7,
		},
```

**Comment by Ezhh** [comment]
Okay, rebased due to conflict and updated noise (I think that looks much better).

**Pull request merged**